packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm

```diff
@@ -98,6 +98,9 @@ @implementation RCTScrollViewComponentView {
   // some other part of the system scrolls scroll view.
   BOOL _isUserTriggeredScrolling;
   BOOL _shouldUpdateContentInsetAdjustmentBehavior;
+  BOOL _automaticallyAdjustKeyboardInsets;
+  BOOL _inverted;
```

Comment:
is there a better way to check for inverted scrollview rather than keeping a state variable that can get out of sync? Perhaps we can check whether there are transformation involved.

Reply:
Im setting this based on the `isInvertedVirtualizedList` prop which also might be incorrect. I will try to refactor it and check based on the transformations

Reply:
So the `inverted` prop gets passed by VirtualizedList when we use FlatList (`inverted` is not on ScrollView). 

It gets passed as the `isInvertedVirtualizedList`: 
https://github.com/facebook/react-native/blob/fdfa0b17bb8c0b55c9c2569e7d3ffcf11c0c5072/packages/virtualized-lists/Lists/VirtualizedList.js#L1088

So this seems like a good solution to handle this case but for some reason the check

```objc
if (oldScrollViewProps.isInvertedVirtualizedList != newScrollViewProps.isInvertedVirtualizedList) {
    _inverted = newScrollViewProps.isInvertedVirtualizedList;
}
```

Never gets called, and the value of `props.isInvertedVirtualizedList` never changes...

I've tried to decode the transformation object and it looks like this style: `{transform: [{scaleY: -1}]}` gets applied but I am not sure how you can decode it on the native side (probably something like this.transform3D which looks really complex..) Maybe you have an idea why this prop value is not changed or do you think its better to go with the transforms? 🤔

Reply:
Okay I found a way to make it work using `transform3D`. There is still another issue that might be caused by something else but it messes up the calculations when the list is `inverted`. 

Here is old arch (as you can see there is no space at the top and end of the scroll view): 

https://github.com/user-attachments/assets/1a319fed-31a4-48d2-a275-34222f705618

And here is new arch (it adds some spacing at the bottom which causes the calculations to be wrong, it's also not happening every time):

https://github.com/user-attachments/assets/d2ab7c53-d093-42f8-9708-95a8b99def97

Im not sure but it seems like a layouting problem which should be solved outside of the scope of this ticket..

Reply:
I agree.. We also have a whole stream of work relate to scrollView and transformations, so we should be good to go!